### PR TITLE
Feature/add address contract distinctor

### PR DESCRIPTION
**Pull request by VargaElod23**

### CODE REVIEW

File: services/explorer/src/hooks/useEthers.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { ethers } from 'ethers';
+import { useMemo } from 'react';
+import { Network } from 'src/utils';
+import { useExplorerNetwork } from './useExplorerNetwork';
+
+function useChain() {
```
Comment (Malak67) [inline services/explorer/src/hooks/useEthers.tsx:6]: Can't we just use from wagmi `useProvider` and `useSigner` ?

File: services/explorer/src/pages/AddressInfo/AddressInfo.effects.tsx [hunk 2]
```diff
@@ -12,6 +12,7 @@ import {
   useGetTransactionsByAddress,
 } from '../../api';
 import { fromWeiToTara } from '../../utils';
+import useChain from 'src/hooks/useEthers';
```
Comment (Malak67) [inline services/explorer/src/pages/AddressInfo/AddressInfo.effects.tsx:15]: This should be relative import

File: services/explorer/src/pages/AddressInfo/AddressInfo.effects.tsx [hunk 3]
```diff
@@ -80,6 +84,7 @@ export const useAddressInfoEffects = (
   });
   const { initLoading, finishLoading } = useExplorerLoader();
   const { backendEndpoint, currentNetwork } = useExplorerNetwork();
+  const { provider } = useChain();
```
Comment (Malak67) [inline services/explorer/src/pages/AddressInfo/AddressInfo.effects.tsx:87]: I suggest to try using wagmi so we don't need another hook.

**Review by Malak67 — APPROVED** [review]

File: services/explorer/src/hooks/useEthers.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        }
+        case Network.DEVNET: {
+          currentProvider = `${process.env.REACT_APP_TARAXA_DEVNET_PROVIDER}`;
+          networkDetails.name = 'Taraxa Devnet';
+          networkDetails.chainId = 843;
+          break;
+        }
+      }
+      console.log(`Provider: ${currentProvider}`);
+      console.log(`Details: ${networkDetails}`);
```
Comment (rattrap) [inline services/explorer/src/hooks/useEthers.tsx:35]: Let's remove these

**Review by rattrap — CHANGES_REQUESTED** [review]

**Review by rattrap — APPROVED** [review]

**Pull request merged**